File: 29_divide_two_integers.py
class Solution:
    def divide(self, dividend: int, divisor: int) -> int:
        
        n=0
        k=divisor
        i=0
        j=0
        if divisor<0:
            i=1
            divisor=-1*divisor
        if dividend<0:
            j=1
            dividend=-1*dividend
        
      
        while divisor<=dividend:
            
            
            n=n+1
            divisor=divisor+k
        result=n
        if i==1 or j==1:
            return (result*-1)
        else:
            return result
            
        
        
            

File: 29_Divide_Two_Integers.cpp
class Solution {
public:
   int divide(int dividend, int divisor) {
       int flag=1;
        
        if(dividend<0 && divisor<0)
        {
            flag=1;
        }
        
        else if(dividend<0)
        {
            
            flag=-1;
        }
        else if(divisor<0)
        {
            
            flag=-1;
        }
       
       long a=labs(dividend);
        long b=labs(divisor);
     
        long result = exp(log(a)-log(b));
        
        if(result >INT_MAX)
            return flag==1 ? INT_MAX:INT_MIN;
        return flag*(int)res;
       
   }
}; 